e2etest/bgp_test.go

```diff
@@ -93,32 +93,42 @@ var _ = ginkgo.Describe("BGP", func() {
 			},
 		},
 	}
-	ginkgo.BeforeEach(func() {
-		var err error
-		if containersNetwork == "host" {
-			if net.ParseIP(hostIPv4) == nil {
-				framework.Fail("Invalid hostIPv4")
-			}
-			if net.ParseIP(hostIPv6) == nil {
-				framework.Fail("Invalid hostIPv6")
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentGinkgoTestDescription().Failed {
+			for _, c := range frrContainers {
+				dump, err := frr.RawDump(c)
+				framework.Logf("External frr dump for %s %s %v", c.Name, dump, err)
 			}
+			DescribeSvc(f.Namespace.Name)
 		}
-		cs = f.ClientSet
-		frrContainers, err = createFRRContainers(containersConf)
-		framework.ExpectNoError(err)
 	})
 
 	ginkgo.AfterEach(func() {
+		ginkgo.By("Clearing the previous configuration")
```

Comment:
I noticed if I ran the test multiple time we endup with lots of svc so we don't clean created service, can we add it here ?

Reply:
That's if you get a failure / it gets interrupted in the middle, then framework doesn't clean the namespace.
I want to have a follow up to ensure external frr containers are wiped out when tests start, I'll try to incorporate that change there.